assets/app/scripts/services/pods.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("PodsService", function($filter) {
+    var getLabel = $filter('label');
+    var debugLabelKey = _.constant('debug.openshift.io/name');
+
+    return {
+      getDebugLabel: function(pod) {
+        return getLabel(pod, debugLabelKey());
+      },
+
+      // Generates a copy of pod for debugging crash loops.
+      generateDebugPod: function(pod, containerName) {
+        var container = _.find(pod.spec.containers, { name: containerName });
+        if (!container) {
+          return null;
+        }
+
+        // Copy the pod and make some changes for debugging. Use the same
+        // metadata as `oc debug`.
+        var debugPod = angular.copy(pod);
+        debugPod.metadata = {
+          name: pod.metadata.name + "-debug",
```

Comment:
we could go over the name length limit but clayton says the CLI isnt handling that case either and to not worry about it right now :)